File: src/main/java/com/zepben/cimbend/database/sqlite/tables/iec61968/common/TableDocuments.java
/*
 * Copyright 2020 Zeppelin Bend Pty Ltd
 * This file is part of evolve-sdk-jvm.
 *
 * evolve-sdk-jvm is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * evolve-sdk-jvm is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with evolve-sdk-jvm.  If not, see <https://www.gnu.org/licenses/>.
 */
package com.zepben.cimbend.database.sqlite.tables.iec61968.common;

import com.zepben.annotations.EverythingIsNonnullByDefault;
import com.zepben.cimbend.database.Column;
import com.zepben.cimbend.database.sqlite.tables.iec61970.base.core.TableIdentifiedObjects;

import static com.zepben.cimbend.database.Column.Nullable.NOT_NULL;
import static com.zepben.cimbend.database.Column.Nullable.NULL;

@EverythingIsNonnullByDefault
public abstract class TableDocuments extends TableIdentifiedObjects {

    public final Column TITLE = new Column(++columnIndex, "title", "TEXT", NOT_NULL);
    public final Column CREATED_DATE_TIME = new Column(++columnIndex, "created_date_time", "TEXT", NULL);
    public final Column AUTHOR_NAME = new Column(++columnIndex, "author_name", "TEXT", NOT_NULL);
    public final Column TYPE = new Column(++columnIndex, "type", "TEXT", NOT_NULL);
    public final Column STATUS = new Column(++columnIndex, "status", "TEXT", NOT_NULL);
    public final Column COMMENT = new Column(++columnIndex, "comment", "TEXT", NOT_NULL);

}

File: src/main/java/com/zepben/cimbend/database/sqlite/tables/iec61968/common/TableDocuments.kt
/*
 * Copyright 2020 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.zepben.cimbend.database.sqlite.tables.iec61968.common

import com.zepben.cimbend.database.Column
import com.zepben.cimbend.database.Column.Nullable.NOT_NULL
import com.zepben.cimbend.database.Column.Nullable.NULL
import com.zepben.cimbend.database.sqlite.tables.iec61970.base.core.TableIdentifiedObjects

@Suppress("PropertyName")
abstract class TableDocuments : TableIdentifiedObjects() {

    val TITLE = Column(++columnIndex, "title", "TEXT", NOT_NULL)
    val CREATED_DATE_TIME = Column(++columnIndex, "created_date_time", "TEXT", NULL)
    val AUTHOR_NAME = Column(++columnIndex, "author_name", "TEXT", NOT_NULL)
    val TYPE = Column(++columnIndex, "type", "TEXT", NOT_NULL)
    val STATUS = Column(++columnIndex, "status", "TEXT", NOT_NULL)
    val COMMENT = Column(++columnIndex, "comment", "TEXT", NOT_NULL)

}
